### ISSUE TITLE
Android build released on F-Droid

### ISSUE THREAD

**Issue by kerpan** [jitsi/jitsi-meet#1290]
It would be great to have the jitsi meet build released on [F-Droid](http://f-droid.org/). Is there any plan to release it there?

**Comment by saghul** [comment]
> On Feb 3, 2017, at 14:54, Davide Michelino <notifications@github.com> wrote:
> 
> It would be great to have the jitsi meet build released on F-Droid <http://f-droid.org/>. Is there any plan to release it there?
> 

Hi,

I don’t think we have considered it.  Do you happen to have a link to some documentation on how the process would go?

--
Saúl

**Comment by kerpan** [comment]
Yes, [this](https://f-droid.org/wiki/page/Inclusion_How-To) is the inclusion how-to and [this](https://f-droid.org/wiki/page/Inclusion_Policy) is the inclusion policy. Would be really great to have it there!

**Comment by saghul** [comment]
Thanks!  I'll check those out.

**Comment by GitanHubert** [comment]
Meet Jitsi on F-Droid : Very good suggestion.

**Comment by lanodan** [comment]
Or even have a link to download the APK. (I had to download it like warez, WTF)

**Comment by Tony-il-Capo** [comment]
> Or even have a link to download the APK. (I had to download it like warez, WTF)

Yes, same here... a very bad process of getting an open source app.

**Comment by robinpaulson** [comment]
Hi, I have asked the F-droid maintainers to include Jitsi in the repository. The issue is here:
https://gitlab.com/fdroid/rfp/issues/243

It would be appreciated if you could assist with any issues which come up, for instance dependencies, binary blobs, build scripts, etc.

Cheers,

Robin

**Comment by saghul** [comment]
@robinpaulson Thanks for doing that Robin! I left a comment in case they need our help.

@lanodan @Tony-il-Capo We currently only provide it via the Play store, but of course you can compile it yourself.

**Comment by lanodan** [comment]
@saghul 
Downloading via Google is not a solution. (specially if you don’t have Google installed)
Building android apps is a horrible mess, specially if your OS/distro is not like debian/ubuntu. (And I’m used to compile things)

Seriously I fail to see how that can be considered okay.

**Comment by saghul** [comment]
@lanodan Time is not infinite, so we do what we can given time constraints. "It never rains to everyone's taste", I'm afraid.

Our currently supported distribution mechanism is the Play Store. Period. F-Droid is in the works, as you read above, but that's out of our control. I did volunteer to help in any capacity required but I haven't been contacted back. You can try to download the APK from one of those APK mirrors, though I cannot recommend one. And last, you can compile it yourself.

If none of the above work for you, I'm sorry, but there is nothing I can do to help you.

**Comment by vn971** [comment]
lanodan, you can also use `gplaycli` on Linux or `yalp` from F-Droid to download apk-s.

@saghul Thanks for your work, I hope F-Droid guys or enthusiasts will reply back in time. I personally stick to F-Droid and have a very, very limited number of apps outside of F-Droid. This is because fdroid is a serious and respected review stage, passing which means the app is compatible with a lot of requirements (_all_ of which I agree to).

**Comment by robinpaulson** [comment]
@saghul I appreciate you getting in touch on the f-droid issues page. As you may have worked out, there is a bit of a backlog for adding apps to f-droid. If you are interested in helping out, there is a process to speed things along, I will see if I can find the links and post them here.

**Comment by saghul** [comment]
@vn971 I didn't know about `gplaycli`, it looks useful, thanks! Also, I don't disagree with anything you said :-)

@robinpaulson Thanks! What kind of help would be needed?

**Comment by robinpaulson** [comment]
@saghul 
How to add the app yourself - actually, anyone can do this.
https://gitlab.com/fdroid/fdroiddata/blob/master/README.md
https://gitlab.com/fdroid/fdroiddata/blob/master/CONTRIBUTING.md

**Comment by saghul** [comment]
@robinpaulson Sweet! I'll give it a try this weekend if time allows!

**Comment by robinpaulson** [comment]
@saghul Are you getting on OK with the inclusion in gitlab? If you need help, do ask!

**Comment by saghul** [comment]
Alas I'm quite swamped right now, so I never got the chance to look at it... yet!

**Comment by saljut7** [comment]
+1

**Comment by labiloute** [comment]
+1 !

**Comment by jpggithub** [comment]
+1

**Comment by sjjh** [comment]
+1

**Comment by mvgorcum** [comment]
+1

**Comment by luke1987515** [comment]
+1

**Comment by lillet23** [comment]
+1

**Comment by SirCmpwn** [comment]
+1

**Comment by vn971** [comment]
Guys, there is a "react with emoji" action instead of +1. Even though I completely agree with the need to have an F-Droid build.

**Comment by narcisgarcia** [comment]
+1

**Comment by ajaydata** [comment]
Try out VideoMeet app from Android a d IOS and see Jitsi implementation

**Comment by robinpaulson** [comment]
> F-Droid is in the works, as you read above, but that's out of our control. I did volunteer
> to help in any capacity required but I haven't been contacted back.

They won't contact you back; there are so many requests and so few maintainers, anything which requires even the tiniest effort beyond an app with a simple process gets put to one side. Not ideal, but their time is limited and there are 276 open requests for app inclusion.

As I understand it, now that you have been made aware of the requirements, they will expect you (or someone else) to make pull requests for the app list repository on gitlab.

I get that your time is limited though, particularly with Play Store, iOS, etc. pressures. Unless someone here prepares the pull requests, it's likely not going to happen.

**Comment by tuxayo** [comment]
@strk 
> please please please. A friend did not want to install jitsi on Android because she doesn't want to give private data. We were disappointed to not find it on f-droid :(

Circumvention: use Yalp Store instead of the Play Store https://f-droid.org/en/packages/com.github.yeriomin.yalpstore/

No Google account needed.

**Comment by narcisgarcia** [comment]
Tried to install Jitsi with Yalp on Cyanogenmod 12.1 but environment tools (Trebuchet, Catapult, Yalp itself) crashed completely.
Luckily I could open a terminal to remove with:
`su
pm list packages | grep jitsi
pm uninstall org.jitsi...`

**Comment by Perflyst** [comment]
Any updates on this?

**Comment by mvglasow** [comment]
I am currently working on a build recipe for F-Droid and I have a few questions:

The instructions at https://github.com/jitsi/jitsi-meet/blob/master/doc/mobile.md tell me to use `react-native run-android` to build the APK, install it on a connected Android device and launch it. How can I just build an unsigned release APK, without installing or launching anything? (And where would I then find the APK?) I have tried cd'ing to the android subdir and running gradle there, but this fails with:
```
* Where:
Build file '/builds/mvglasow/Navit-fdroid/fdroiddata/build/org.jitsi.meet/android/build.gradle' line: 73

* What went wrong:
A problem occurred configuring project ':react-native-background-timer'.
> Failed to notify project evaluation listener.
   > /builds/mvglasow/Navit-fdroid/fdroiddata/build/org.jitsi.meet/node_modules/react-native-background-timer/package.json (No such file or directory)
```

(Tested with 10624e87.)

Also, how can I tell which git commits correspond to an official release (i.e. one that would go on the Play Store)? Are they tagged in some way? I see two types of tags, `nnnn` (number) and `jitsi-meet_nnnn`, do any of those tags correspond to releases? Or is there another way to tell automatically?

**Comment by Rudloff** [comment]
> /builds/mvglasow/Navit-fdroid/fdroiddata/build/org.jitsi.meet/node_modules/react-native-background-timer/package.json (No such file or directory)

It looks like NPM dependencies were not installed.

You might want to have a look at how we build other React Native apps on F-Droid.
We don't use the `react-native` command and instead do something like this manually:
```bash
npm install
cd android
gradle assemble
```
(But Jitsi Meet might have some additional build steps.)

**Comment by mvglasow** [comment]
@Rudloff great to see you here! Do you happen to have any examples of other React Native apps on F-Droid? (I wasn’t even aware Jitsi Meet was not the only one…)

**Comment by Rudloff** [comment]
You can check the following build recipes:
* com.lesspass.android
* de.fragdenstaat.app
* com.mdroid

**Comment by saghul** [comment]
For all intents and purposes our use of React Native can be counted as anecdotal. It's an implementation detail. You can build it as a regular Android package. The extra step you need to fllow is installing all dependencies with `npm i`.

If you make a debug build, however, you'd need to start the React Native packager, but you don't need that.

We currently have no tag for mobile releases but are about to start doing it (next release will have one).

**Comment by mvglasow** [comment]
Anecdotal… well, you do need React Native installed to build it :-)

Re tags for mobile releases, will they follow a particular pattern? F-Droid can do some regex-based filtering, so if there were e.g. a prefix used for all mobile releases (and only for those), that would be quite easy.

**Comment by mvglasow** [comment]
@Rudloff I tried one of those build recipes. I had to run `npm install` with sudo or else it would fail. The `truncate`/`rm` operations fail because the respective files are not found, so I just appended `|| true` to see what happens afterwards. After that, the build process fails with the same error as above.

This might be related to my CI setup, I’ve documented details at https://gitlab.com/mvglasow/fdroiddata/issues/1 and would be grateful if you could take a look.

**Comment by Rudloff** [comment]
I just noticed the app uses several non-free libraries: https://github.com/jitsi/jitsi-meet/blob/54df0f5d0fe1565c93118df8aa1bc981dd738c43/android/app/build.gradle#L85

This will prevent the app from being included in F-Droid as we require included libraries to be FOSS.

**Comment by mvglasow** [comment]
Is there any way to build Jitsi Meet without FCM? If not, that would be a blocker.

**Comment by saghul** [comment]
> Is there any way to build Jitsi Meet without FCM? If not, that would be a blocker.

Yes and no. FCM is not a hard requirement, so a build time toggle to disable it would be ok. I wonder if we can do it with this variable: https://github.com/jitsi/jitsi-meet/blob/6190173ea74be8f070f9a8334ce699f22570ac51/android/app/build.gradle#L3

Then here: https://github.com/jitsi/jitsi-meet/blob/master/android/app/src/main/java/org/jitsi/meet/MainActivity.java#L194 we'll need to use the full package notation to avoid the imports.

I think that would be all.

I'll keep this in mind and take care of it as time allows, but if someone wants to send a PR, that would be most welcome.

**Comment by webmind** [comment]
Would be very welcome for me as well, yalp needs a google account here and gplaycli don't work for me on ubuntu bionic and I've got a google-free smartphone. 

Thanks!

**Comment by Neustradamus** [comment]
I think it must be resolved before the release (vulnerabilities):
- https://github.com/jitsi/jitsi-universe-public/issues/1
- https://github.com/jitsi/jitsi/issues/562

**Comment by saghul** [comment]
@Neustradamus Those have nothing to do with Jitsi Meet. Jitsi Meet doesn't use Bouncy Castle.

**Comment by vn971** [comment]
@webmind offtopic, but Yalp certainly doesn't require a personal google account. There were some occasions when it was laggy, but otherwise it works just fine (using built-in accounts). Retrying later helps at times.

**Comment by rasos** [comment]
Could we have an F-Droid version without Google Crashlytics and Firebase Analytics? See https://reports.exodus-privacy.eu.org/en/reports/37219/

**Comment by saghul** [comment]
I've made some progress on this. https://github.com/jitsi/jitsi-meet/pull/4155 allows to build Jitsi Meet without any analytics nor Google Play Services.

If I can get this landed by in the next Jitsi Meet version (19.2.0) it would be the right time to restart the F-Drop release effort, by taking that tag and rebuilding it with the `LIBRE_BUILD` env var.

**Comment by raphj** [comment]
For reference: https://gitlab.com/fdroid/rfp/issues/73#note_188026138

**Comment by PanderMusubi** [comment]
See also https://github.com/jitsi/jitsi-meet/issues/4588

**Comment by saghul** [comment]
https://f-droid.org/en/packages/org.jitsi.meet/ 🚀

**Comment by PanderMusubi** [comment]
I think, this is one to announce on social media, reddit, irc, etc.

**Comment by PanderMusubi** [comment]
See also https://github.com/jitsi/jitsi-meet/issues/4891

**Comment by heyakyra** [comment]
It seems the f-droid package is very behind the play store version.

**Comment by saghul** [comment]
It isn't. 22.7.1 is the latest stable version on the store. 23.0.0 is still in testing.

**Comment by Yolakalemowa** [comment]
> It isn't. 22.7.1 is the latest stable version on the store. 23.0.0 is still in testing.

Hi... is jitsi for android not maintained anymore?

**Comment by saghul** [comment]
Of course it is. We just released 24.3.0 and F-Droid usually takes a bit to incorporate it. They have 24.2.2 still: https://f-droid.org/en/packages/org.jitsi.meet/

What makes you think it's not maintained anymore?!

**Comment by Yolakalemowa** [comment]
> Of course it is. We just released 24.3.0 and F-Droid usually takes a bit to incorporate it. They have 24.2.2 still: https://f-droid.org/en/packages/org.jitsi.meet/
> 
> What makes you think it's not maintained anymore?!

oh sorry I saw [this](https://github.com/jitsi/jitsi-android#unmaintained) and got confused. tbh I'm not an expert in Github.

If I may ask you, is the APK available elsewhere other than f-droid? I use Obtanium on Graphene OS cause it seems to have better security than F-Droid. And even better, would it be possible to provide it via Accrescent when they start accepting submissions? Accrescent would be the most ideal option for GOS users.

**Comment by saghul** [comment]
Oh, that! Good point, I'll add some pointer in the README.

Currently we suppport F-Droid and the Play Store. So far you are the 2nd person to ask about Accrescent, we'll see how that evolves and consider it for the future.
